abi/src/test/java/org/web3j/abi/TypeDecoderTest.java

```diff
@@ -858,11 +859,63 @@ public void testIntDecode() throws Exception {
                         Int256.class),
                 (new Int256(BigInteger.valueOf(-1))));
 
+
+
+
+        assertEquals(
+                TypeDecoder.decodeNumeric(
+                        TypeEncoder.encodeNumeric(new Int256(BigInteger.TWO.pow(248))),
+                        Int256.class),
+                new Int256(BigInteger.TWO.pow(248)));
+
+        assertEquals(
+                TypeDecoder.decodeNumeric(
+                        TypeEncoder.encodeNumeric(new Int256(BigInteger.TWO.pow(248).negate().subtract(BigInteger.ONE))),
+                        Int256.class),
+                new Int256(BigInteger.TWO.pow(248).negate().subtract(BigInteger.ONE)));
+
+
+
+        assertEquals(
+                TypeDecoder.decodeNumeric(
+                        "0x7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff",
+                        Int256.class),
+                new Int256(new BigInteger("57896044618658097711785492504343953926634992332820282019728792003956564819967")));
+
+        assertEquals(
+                TypeDecoder.decodeNumeric(
+                        "0x8000000000000000000000000000000000000000000000000000000000000000",
+                        Int256.class),
+                new Int256(new BigInteger("-57896044618658097711785492504343953926634992332820282019728792003956564819968")));
+
         assertEquals(TypeDecoder.instantiateType("int", 123), (new Int(BigInteger.valueOf(123))));
 
         assertEquals(TypeDecoder.instantiateType("int", -123), (new Int(BigInteger.valueOf(-123))));
     }
 
+    @Test
+    public void testInt16All() throws Exception {
```

Comment:
pick one type that won't take too long to run test to test the entire range for sanity

Reply:
So `2^255 - 1` and `-2^255` used to fail in decoding?

Reply:
testing all Int16 and uint16 can be a overkill, add tests for boundary cases should be enough.
What do you think  @gtebrean ?

Reply:
> testing all Int16 and uint16 can be a overkill, adding tests for boundary cases should be enough. What do you think @gtebrean ?

it is a loop with 2^16 = 65536 entries, I think it runs at a reasonable time. But if there is concern I can change it to boundary

Reply:
> So `2^255 - 1` and `-2^255` used to fail in decoding?

yes, that's the case I'm using, later on found that the bug is in a large range mentioned in PR title

Reply:
done